src/lesson2-homework.ts

```diff
@@ -54,58 +66,101 @@ excercise6();
+  const printGreeting = (user: Tuser): string => user.name;
+  type Tproduct = { name: string; price: number };
+  let product: Tproduct = {
+    name: 'Bread',
+    price: 11,
+  };
+  console.log(printGreeting(product));
+  console.log(printGreeting({ name: 'Bread', price: 11 }));
+  product.color = 'white';
+  type Product = {
+    name: string;
+    price: number;
+    color: string;
+  };
+  const product2: Product = {
+    name: 'Bread',
+    price: 11,
+    color: 'white',
+  };
+  product = product2;
+  console.log(product);
 }
-// TODO: compile and run the code
 excercise8();
 
 // declare a `Book` class with a constructor and a method
 function excercise9() {
-  // TODO: declare a `Book` class with a constructor and a method `getInfo` which returns the book info as a string
-  // TODO: constructor should take three parameters - title and year of publication
-  // TODO: method `getInfo` should return the book title and year as a string
-  // TODO: create a book object and call the method `getInfo`, print the result to console
-  // TODO: assign a new value to the year property
-  // TODO: call the method `getInfo` again
-  // TODO: add a new method `getAge` which returns the age of the book (current year - year of publication)
-  // TODO: call the method `getAge` and print the result to console
-  // TODO: add a new method `revise` which takes a new year as a parameter and updates the year property, add validation to the method - year can not be in the future, year can not be less than prev year
-  // TODO: call the method `revise` and pass a new year as a parameter
-  // TODO: add private modifier to the year property
-  // TODO: try to access the year property from outside of the class - observe the error
-  // TODO: change protected modifier to the year property, remove private modifier
-  // TODO: create a subclass `Magazine` which extends `Book` class
-  // TODO: add a new properties `month` and `day` to the `Magazine` class
-  // TODO: add constructor override to the Magazine class which takes four parameters - title, year, month and day
-  // TODO: use super keyword to call the `Book` class constructor with title and year parameters
-  // TODO: add a method override `getInfo` to the `Magazine` class which prints the magazine info to console
-  // TODO: use super keyword to call the `getInfo` method of the `Book` class
-  // TODO: create a magazine object and call the method `getInfo`, print the result to console
-  // TODO: call the inherited method `getAge` of the magazine object and print the result to console
+  class Book {
+    title: string;
+    year: number;
+    constructor(title: string, year: number) {
+      this.title = title;
+      this.year = year;
+    }
+    getInfo(): string {
+      return `${this.title} is published in ${this.year}`;
+    }
+    getAge() {
+      return new Date().getFullYear() - this.year;
+    }
+    revise(year: number) {
+      if (
+        year > new Date().getFullYear() ||
```

Comment:
maybe store the current year in a separate variable to prevent calling same w Date().getFullYear()  twice